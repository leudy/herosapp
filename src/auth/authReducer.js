import { types } from "../types/authTypes";

export const authReducer = (state = {}, action) => {


  
  switch (action.type) {
    case types.LOGIN: {
      const Data = { ...action.payload, logged: true };
      localStorage.setItem("user", JSON.stringify(Data));
      return Data;
    }
    case types.LOGOUT: {
      localStorage.removeItem("user");
      return { logged: false };
    }

    default:
      return state;
  }
};
