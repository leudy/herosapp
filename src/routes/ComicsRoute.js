import React from "react";
import { Route, Redirect, Switch } from "react-router-dom";
import { MarvelScreen } from "../marvel/MarvelScreen";
import { Navbar } from "../shared/Navbar";
import { DcScreen } from "../dc/DcScreen";
import { HeroeScreen } from "../heroes/HeroeScreen";
import { SearchComponent } from "../search/SearchComponent";

export const ComicsRoute = () => {
  //console.log("comics route was load");
  return (
    <>
      <Navbar></Navbar>
      <div className="container">
        <Switch>
          <Route exact path="/marvel" component={MarvelScreen} />
          <Route exact path="/heroe/:heroeid" component={HeroeScreen} />
          <Route exact path="/dc" component={DcScreen} />
          <Route exact path="/search" component={SearchComponent} />
          <Redirect to="/marvel" />
        </Switch>
      </div>
    </>
  );
};
