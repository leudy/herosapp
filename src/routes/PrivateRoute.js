import React from "react";
import PropTypes from "prop-types";
import { Redirect, Route } from "react-router-dom";
export const PrivateRoute = ({
  IsAuthenticated,
  component: Component,
  ...rest
}) => {
  localStorage.setItem("lastpath", rest.location.pathname);
  if (rest.location.search.lenght > 2) {
    localStorage.setItem("param", rest.location.search);
    //console.log(rest.location.search);
  }

  return (
    <Route
      {...rest}
      component={(props) =>
        IsAuthenticated ? <Component {...props} /> : <Redirect to="/login" />
      }
    />
  );
};

PrivateRoute.propTypes = {
  IsAuthenticated: PropTypes.bool.isRequired,
  component: PropTypes.func.isRequired,
};
