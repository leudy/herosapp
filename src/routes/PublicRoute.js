import React from "react";
import PropTypes from "prop-types";
import { Redirect, Route } from "react-router-dom";
export const PublicRoute = ({
  IsAuthenticated,
  component: Component,
  ...rest
}) => {
  return (
    <Route
      {...rest}
      component={(props) =>
        !IsAuthenticated ? <Component {...props} /> : <Redirect to="/" />
      }
    />
  );
};

PublicRoute.propTypes = {
  IsAuthenticated: PropTypes.bool.isRequired,
  component: PropTypes.func.isRequired,
};
