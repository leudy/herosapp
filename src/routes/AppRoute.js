import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { LoginScreen } from "../Login/LoginScreen";
import { ComicsRoute } from "./ComicsRoute";

export default function AppRoute() {
  return (
    <Router>
      <div>
        <Switch>
          <Route exact path="/login" component={LoginScreen}></Route>
          <Route path="/" component={ComicsRoute}></Route>
        </Switch>
      </div>
    </Router>
  );
}
