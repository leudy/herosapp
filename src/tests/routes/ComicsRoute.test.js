import React from "react";
import Adapter from "enzyme-adapter-react-16"
import {mount, configure} from "enzyme";    
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from "../../auth/AuthContext";
import { ComicsRoute } from "../../routes/ComicsRoute";
configure({adapter: new Adapter()});


describe('Probando el Comics Route', () => {
    const _contextValues  ={
        dispatch:jest.fn(),
        user:{
            logged:true,
            name:"Pedro Juan Varon"
             
        }
    }
    test('should be display correctly', () => {
        const wrapper = mount(
            <MemoryRouter>
                 <AuthContext.Provider value={_contextValues}> 
                <ComicsRoute></ComicsRoute>
            </AuthContext.Provider>
            </MemoryRouter>
           
              );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(".text-info").text().trim()).toBe(_contextValues.user.name);

    })
    
})
