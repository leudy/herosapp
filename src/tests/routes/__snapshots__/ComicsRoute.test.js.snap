// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el Comics Route should be display correctly 1`] = `ReactWrapper {}`;
